app/internal/security/ouath2.py

```diff
@@ -0,0 +1,117 @@
+from datetime import datetime, timedelta
+from typing import Union
+
+from app.config import JWT_ALGORITHM, JWT_KEY, JWT_MIN_EXP
+from passlib.context import CryptContext
+from app.database.models import User
+from fastapi import Depends, HTTPException
+from fastapi.security import OAuth2PasswordBearer
+import jwt
+from jwt.exceptions import InvalidSignatureError
+from sqlalchemy.orm import Session
+from starlette.requests import Request
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_401_UNAUTHORIZED
+from .schema import LoginUser
+
+
+pwd_context = CryptContext(schemes=["bcrypt"])
+oauth_schema = OAuth2PasswordBearer(tokenUrl="/login")
+
+
+async def get_db_user_by_username(db: Session, username: str) -> User:
```

Comment:
I believe this function should be inside the `User` model, or on `/app/internal/users.py`, and named `by_username`.

Keep it simple and think about other developers who want to use it :)

Reply:
I know it does.
The reason i have it there is because i though that it will be 'more corrects' to build the login and authentication systems as a stand-alone system. A code that needs the minimum changes if someone wants to use it in another project.
I will delete it if you prefer, and use the other function.

Reply:
I do prioritize DRY code in this case :)